### PR TITLE
Adding Module and Example for ECS cluster monitoring with ecs_observer

### PR DESCRIPTION
**Pull request by ruchimo**
### What does this PR do?

Adding Module and Example for ECS cluster monitoring with ecs_observer

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted. Consult the CONTRIBUTING guide for submitting pull-requests.

### Motivation

Part of the virtual Hackathon on the [AWS Observability Accelerator](https://github.com/aws-observability/terraform-aws-observability-accelerator).

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I ran `pre-commit run -a` with this PR
- [x] Yes, I have added a new example under [examples](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/examples) to support my PR (when applicable)
- [x] Yes, I have updated the [Pages](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/docs) for this feature

**Note**: Not all the PRs required examples and docs.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by bonclay7 — COMMENTED** [review]
### examples/ecs-cluster-with-vpc
Normally we try to keep examples as lean as possible to simplify maintenance. Normally `ecs-cluster-with-vpc` should be flat and not nested. We usually have an apply-able main.tf in the examples. Here's it's not clear what we are testing. 
Additionally there's a quite popular module actively maintained, maybe we should just reuse it and focus on our added value? https://github.com/terraform-aws-modules/terraform-aws-ecs but maybe I'm missing something?

File: modules/ecs_monitoring/README.md
Comment (bonclay7) [inline modules/ecs_monitoring/README.md:1]: Can you rename the module `ecs-monitoring` instead?

**Comment by bonclay7** [comment]
When we finish the example, that use the new module, we should add a little doc https://aws-observability.github.io/terraform-aws-observability-accelerator/ on ECS monitoring as well

File: modules/ecs_monitoring/README.md
Comment (ruchimo) [reply modules/ecs_monitoring/README.md:1]: Done.

File: modules/ecs-monitoring/versions.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.1.0"
```
Comment (bonclay7) [inline modules/ecs-monitoring/versions.tf:2]: Can we try version 1.0.0 as the min? we will update our tests if we have to

File: modules/ecs-monitoring/configs/config.yaml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        new_name: NetworkTxBytes
+      - include: ecs.task.storage.read_bytes
+        action: update
+        new_name: StorageReadBytes
+      - include: ecs.task.storage.write_bytes
+        action: update
+        new_name: StorageWriteBytes
+
+exporters:
+  prometheusremotewrite:
```
Comment (bonclay7) [inline modules/ecs-monitoring/configs/config.yaml:113]: can we add the `region` and `role_arn` to basically support cross region and cross account scenarios?

Comment (bonclay7) [reply modules/ecs-monitoring/configs/config.yaml:113]: still missing region here

File: modules/ecs-monitoring/locals.tf [hunk 3]
```diff
@@ -6,4 +6,28 @@ locals {
...
   name             = "amg-ex-${replace(basename(path.cwd), "_", "-")}"
   description      = "AWS Managed Grafana service for ${local.name}"
+
+  default_otel_values = {
+    aws_region                      = data.aws_region.current.name
+    cluster_name                    = var.aws_ecs_cluster_name
+    cluster_region                  = data.aws_region.current.name
+    refresh_interval                = "60s"
+    ecs_metrics_collection_interval = "15s"
+    amp_remote_write_ep             = "https://aps-workspaces.us-east-1.amazonaws.com/workspaces/ws-f9a9b3d8-511e-4640-9b2d-15fbd53f7209/api/v1/remote_write"
```
Comment (bonclay7) [inline modules/ecs-monitoring/locals.tf:16]: I'm not sure you should expose your workspace url like this. Maybe you could create a workspace as part of the example and put the remote write url as an output?

File: modules/ecs-monitoring/locals.tf [hunk 4]
```diff
@@ -6,4 +6,28 @@ locals {
...
+    otlpGrpcEndpoint                = "0.0.0.0:4317"
+    otlpHttpEndpoint                = "0.0.0.0:4318"
+  }
+
```
Comment (bonclay7) [inline modules/ecs-monitoring/locals.tf:20]: I think all of those should be `ecs-monitoring` module's attribute

File: modules/ecs-monitoring/locals.tf [hunk 5]
```diff
@@ -6,4 +6,28 @@ locals {
...
+  ssm_param_value = yamlencode(
+    templatefile("${path.module}/configs/config.yaml", local.default_otel_values)
+  )
+
+  container_def_default_values = {
+    container_name = "adot_new"
+    otel_image_ver = "v0.31.0"
```
Comment (bonclay7) [inline modules/ecs-monitoring/locals.tf:27]: This as well should be a module variable

File: modules/ecs-monitoring/locals.tf (hunk 3 above)
Comment (bonclay7) [reply modules/ecs-monitoring/locals.tf:16]: You could also as part of the module, have a `enable_managed_prometheus = true` that creates an amp workspace

**Comment by bonclay7** [comment]
@ruchimo any updates on this?

**Comment by bonclay7** [comment]
@ruchimo please let me know when I ca re-review

File: modules/ecs-monitoring/configs/config.yaml (hunk 2 above)
Comment (ruchimo) [reply modules/ecs-monitoring/configs/config.yaml:113]: Can we take this up later?

**Comment by ruchimo** [comment]
@bonclay7 - Please review....

**Comment by bonclay7** [comment]
@ruchimo please check the broken tests

https://github.com/aws-observability/terraform-aws-observability-accelerator/actions/runs/6048526849/job/16418900113?pr=211
https://github.com/aws-observability/terraform-aws-observability-accelerator/actions/runs/6048526849/job/16418897812?pr=211

Also please install and run pre-commit, you can find instructions [here](https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/CONTRIBUTING.md)

**Comment by ruchimo** [comment]
> @ruchimo please check the broken tests
> 
> https://github.com/aws-observability/terraform-aws-observability-accelerator/actions/runs/6048526849/job/16418900113?pr=211 https://github.com/aws-observability/terraform-aws-observability-accelerator/actions/runs/6048526849/job/16418897812?pr=211
> 
> Also please install and run pre-commit, you can find instructions [here](https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/CONTRIBUTING.md)

Checking @bonclay7 , I already have it installed...

**Comment by ruchimo** [comment]
@bonclay7 - I have fixed the failures detected by pre-commit

**Comment by ruchimo** [comment]
@bonclay7 I have checked the failures
1. https://github.com/aws-observability/terraform-aws-observability-accelerator/actions/runs/6070146490/job/16486224503?pr=211 --> This is reposting Potential problems on the Modules we are reusing.
2. https://github.com/aws-observability/terraform-aws-observability-accelerator/actions/runs/6070146491/job/16486220400?pr=211 --> This is from the docs, the link would be accesible once the PR is merged.

**Review by bonclay7 — APPROVED** [review]

**Pull request merged**